packages/asset-service/src/generateAssetData/exetendErc20.ts

```diff
@@ -0,0 +1,62 @@
+import { caip19 } from '@shapeshiftoss/caip'
+import { ChainTypes, ContractTypes, NetworkTypes, TokenAsset } from '@shapeshiftoss/types'
+import axios, { AxiosInstance, AxiosResponse } from 'axios'
+const axiosInstance: AxiosInstance = axios.create()
+
+type YearnApiVault = {
+  inception: number
+  address: string
+  symbol: string
+  name: string
+  display_name: string
+  icon: string
+  token: {
+    name: string
+    symbol: string
+    address: string
+    decimals: number
+    display_name: string
+    icon: string
+  }
+  tvl: {
+    total_assets: number
+    price: number
+    tvl: number
+  }
+  apy: {
+    net_apy: number
+  }
+  endorsed: boolean
+  version: string
+  decimals: number
+  type: string
+  emergency_shutdown: boolean
+}
+
+export const extendErc20 = async (): Promise<TokenAsset[]> => {
+  const response: AxiosResponse = await axiosInstance.get(
```

Comment:
give this a type on the get request, like `get<YearnApiVault[]>(...` and you can destructure data straight off that.

also, we shouldn't try to cast and return `{}` as a TokenAsset, if we don't think it's a vault we should do a filter pass first.

lastly, don't cast `as TokenAsset`, do something like `const result: TokenAsset = {...` and let the compiler tell you what it needs to be

Reply:
Good call. I've updated it.